pfam_experiment.py

```diff
@@ -46,186 +48,439 @@
-                                                              test_family_accessions=lens_knn_train_family_accessions,
-                                                              batch_size=FLAGS.batch_size,
-                                                              train_samples=FLAGS.knn_train_samples)[0]
-	train_knn_accuracy = train_knn_results['1-nn accuracy']
-
-	test_knn_results = pfam_nearest_neighbors_classification(encoder=embedding_optimizer.target, 
-                                                             train_family_accessions=knn_test_family_accessions, 
-                                                             test_family_accessions=knn_test_family_accessions,
-                                                             batch_size=FLAGS.batch_size,
-                                                             train_samples=FLAGS.knn_train_samples)[0]
-	test_knn_accuracy = test_knn_results['1-nn accuracy']
-
-	datum = {
-				'encoder_fn_name': FLAGS.encoder_fn_name,
-				'encoder_fn_kwargs_path': FLAGS.encoder_fn_kwargs_path,
-				'reduce_fn_name': FLAGS.reduce_fn_name,
-				'reduce_fn_kwargs_path': FLAGS.reduce_fn_kwargs_path,
-				'epochs': FLAGS.epochs,
-				'batch_size': FLAGS.batch_size,
-				'encoder_lr': FLAGS.learning_rate[0],
-				'lens_lr': FLAGS.learning_rate[1],
-				'predictor_lr': FLAGS.learning_rate[2],
-				'encoder_wd': FLAGS.weight_decay[0],
-				'lens_wd': FLAGS.weight_decay[1],
-				'predictor_wd': FLAGS.weight_decay[2],
-				'train_families': FLAGS.train_families,
-				'use_transformer': FLAGS.use_transformer,
-				'use_bert': FLAGS.use_bert,
-				'restore_transformer_dir': FLAGS.restore_transformer_dir,
-				'knn_train_samples': FLAGS.knn_train_samples,
-				'lens_cross_entropy': lens_cross_entropy,
-				'lens_accuracy': lens_accuracy,
-				'train_knn_accuracy': train_knn_accuracy,
-				'test_knn_accuracy': test_knn_accuracy
-			}
-	
-	print(datum)
-	df = pd.DataFrame([datum])
-    
-	with gcsfs.open(os.path.join('sweep_data', 'experiment' + '.csv'), 'w') as gcs_file:
-		df.to_csv(gcs_file)
+    if FLAGS.use_transformer:
+        assert (
+            FLAGS.encoder_fn_name == 'transformer'
+        ), 'encoder_fn_name must be transformer if use_transformer is True!'
+
+    assert (FLAGS.epochs % FLAGS.measurements == 0
+            ), 'Number of measurements must divide number of epochs!'
+    measurement_epochs = FLAGS.epochs // FLAGS.measurements
+
+    assert FLAGS.results_save_dir != '', 'Specify results_save_dir!'
+
+    assert FLAGS.label != '', 'Specify label!'
+
+    if FLAGS.load_model:
+        assert FLAGS.load_model_dir != '', 'Specify load_model_dir!'
+        assert FLAGS.load_model_step > 0, 'Loaded model must have been trained for more than 0 steps.'
+
+    if FLAGS.save_model:
+        assert FLAGS.save_model_dir != '', 'Specify save_model_dir!'
+
+    datum = {
+        'label': FLAGS.label,
```

Comment:
not necessary to change, but this would have been more concise if you did something like

_FLAG_VALUES_TO_SAVE = ['label', 'encoder_fn_name', ...]
datum = {key: FLAGS.getattr(key) for key in_FLAG_VALUES_TO_SAVE}

Reply:
thanks for the heads up, will think about that